flask_hw5/blueprints/rooms/rooms_routes.py

```diff
@@ -0,0 +1,57 @@
+from flask_restful import Resource
+from sqlalchemy.exc import SQLAlchemyError
+
+from blueprints.rooms.room_arg_parser import (room_parser,
+                                              room_parser_post,
+                                              room_parser_patch)
+from blueprints.rooms.room_object_serlzer import marshal_with_decor
+from blueprints.rooms.rooms_model import Room
+from hotel_db import db
+
+
+class Rooms(Resource):
+    @marshal_with_decor
+    def get(self):
+        room_filters = room_parser.parse_args()
+        if any(tuple(room_filters.values())):
+            return Room.query.filter_by(**{k: v
+                                           for k, v in room_filters.items()
+                                           if v}).first_or_404()
+        return Room.query.all(), 200
+
+    @marshal_with_decor
+    def post(self):
+        new_room_values = room_parser_post.parse_args()
+        room_to_add = Room(**new_room_values)
```

Comment:
an exception may be here if you pass unknown keys? or do not pass any of required keys?

Reply:
custom parser catch all of those. No exceptions occurred in cases like: empty fields, wrong fields, additional unknown fields, fields with wrong data types etc. Of course, we can implement more precise validation with flask_form and wtf_validators(I am going to do it in my flask project), but I am not sure it is needed in  tasks like this one.